### PR TITLE
feat(compare versions): Highlight the differences between versions

### PR DESCRIPTION
**Pull request by ioanpocol**

### CODE REVIEW

File: scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js [hunk 1]
```diff
@@ -59,6 +72,109 @@ class LinkFunction {
...
+
+    /**
+     * @ngdoc method
+     * @name sdCompareVersionsArticle#setVersionsDifference
+     * @param {Object} item - current item version
+     * @param {Object} oldItem - old item to compare with
+     * @description Changes 'headline', 'abstract', 'body_footer', 'body_html', 'byline' from item
+     * in order to highlight the differences from oldItem
+     */
+    setVersionsDifference(item, oldItem) {
```
Comment (petrjasek) [inline scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js]: imo would be better to return a diff and not modify the item, otherwise you have to keep a copy somewhere

File: scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js [hunk 2]
```diff
@@ -59,6 +72,109 @@ class LinkFunction {
...
+
+        let text = this.diffMatchPatch
+            .diff_prettyHtml(diffs)
+            .replace(/&lt;/g, '<')
+            .replace(/&gt;/g, '>')
+            .replace(/<span><p><\/span>/g, '<p>')
+            .replace(/<span><\/p><\/span>/g, '</p>')
+            .replace(/&nbsp;/g, ' ');
+
+        if (this.config.features.editor3) {
```
Comment (petrjasek) [inline scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js]: why is the diff editor specific?

File: scripts/core/editor3/directive.js [hunk 3]
```diff
@@ -111,6 +115,14 @@ class Editor3Directive {
 
         const store = createStore(this);
 
+        $scope.$watch('vm.value', (newValue, oldValue) => {
```
Comment (petrjasek) [inline scripts/core/editor3/directive.js]: why is this needed?

File: scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js (hunk 1 above)
Comment (ioanpocol) [inline scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js]: Is it already a copy

File: scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js (hunk 2 above)
Comment (ioanpocol) [inline scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js]: Editor3 don't support 'ins' tag and we used 'code' instead.

File: scripts/core/editor3/directive.js (hunk 3 above)
Comment (ioanpocol) [inline scripts/core/editor3/directive.js]: To detect if the value sent to editor3 wrapper directive has been changed and then to update the editor3 state.

File: scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js (hunk 2 above)
Comment (ioanpocol) [inline scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js]: We will move this code on Editor3. @gbbr

File: scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js (hunk 1 above)
Comment (petrjasek) [inline scripts/apps/authoring/compare-versions/directives/CompareVersionsArticleDirective.js]: that's what I mean

File: scripts/core/editor3/directive.js (hunk 3 above)
Comment (petrjasek) [inline scripts/core/editor3/directive.js]: so it's using editor to render the diff?

Comment (ioanpocol) [inline scripts/core/editor3/directive.js]: yes

Comment (petrjasek) [inline scripts/core/editor3/directive.js]: why?

Comment (ioanpocol) [inline scripts/core/editor3/directive.js]: It was more simpler

Comment (petrjasek) [inline scripts/core/editor3/directive.js]: imo that's confusing, it shouldn't be editable right?
wouldn't it work with just `ng-bind-html` ? why is it simpler with editor?

Comment (ioanpocol) [inline scripts/core/editor3/directive.js]: When I started to work on it it was already using the editor, I just added the highlights. The editor is readonly.

Comment (petrjasek) [inline scripts/core/editor3/directive.js]: I see
